packages/mml-viewer/src/StandaloneViewer.ts

```diff
@@ -38,7 +38,37 @@ export class StandaloneViewer {
       event.data.params !== null;
     if (isParamUpdate) {
       this.updateUrlParams(event.data.params);
+      return;
+    }
+
+    const isScreenshot = event?.data?.type === "screenshot" && event.data.imageUri === null;
+    if (isScreenshot) {
+      this.handleScreenshot(event);
+      return;
+    }
+  }
+
+  private handleScreenshot(event: MessageEvent) {
+    try {
+      const imageUri = this.captureScreenshot();
+      (event.source as Window).postMessage(
+        {
+          type: "screenshot",
+          imageUri: imageUri || null,
+        },
+        event.origin,
+      );
+    } catch (error) {
+      console.error("Error capturing screenshot:", error);
+    }
+  }
+
+  private captureScreenshot(): string | null {
+    const canvas = window.document.querySelector("canvas") as HTMLCanvasElement | null;
```

Comment:
This is a rather loose way to find the correct element. It would be better to wire this up through the renderer to make the implementation more resilient.